tooling/apt/src/main/java/org/apache/camel/tools/apt/EndpointAnnotationProcessor.java

```diff
@@ -359,12 +359,22 @@ protected ComponentModel findComponentProperties(RoundEnvironment roundEnv, UriE
         model.setLenientProperties(uriEndpoint.lenientProperties());
         model.setAsync(implementsInterface(processingEnv, roundEnv, endpointClassElement, "org.apache.camel.AsyncEndpoint"));
 
+        String deprecationNote = null;
         // what is the first version this component was added to Apache Camel
         String firstVersion = uriEndpoint.firstVersion();
-        if (Strings.isNullOrEmpty(firstVersion) && endpointClassElement.getAnnotation(Metadata.class) != null) {
-            // fallback to @Metadata if not from @UriEndpoint
-            firstVersion = endpointClassElement.getAnnotation(Metadata.class).firstVersion();
+        Metadata[] metadataArray = endpointClassElement.getAnnotationsByType(Metadata.class);
+        for(Metadata metadata : metadataArray) {
```

Comment:
code formatting looks like a checkstyle problem here

Reply:
there was not checkstyle issue AFAIR. will check again.

Reply:
@onderson , please note, that right now -Psourcecheck option doesn't stop build and just print audit message warning and you need to check it manually.